components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/HttpRequestOperation.java

```diff
@@ -311,7 +312,7 @@ public void onNext(ByteBuf chunk) {
                         } else {
                             String channelIdentifier = String.format("%s -> %s", nettyConnection.channel().localAddress(), nettyConnection.channel().remoteAddress());
                             LOGGER.error(format("Failed to send request body data. origin=%s connection=%s request=%s",
-                                    nettyConnection.getOrigin(), channelIdentifier, request), future.cause());
+                                    nettyConnection.getOrigin(), channelIdentifier, formatRequest(request)), future.cause());
```

Comment:
No need to log the headers here. The exception will eventually propagate to `HttpErrorStatusCauseLogger` which alone can format the request headers.

Reply:
fixed